modules/representation_management/spec/requests/v0/pdf_generator_2122_spec.rb

```diff
@@ -59,29 +60,166 @@
       }
     end
 
-    context 'when submitting all required data' do
-      it 'responds with a created status' do
-        post(base_path, params:)
-        expect(response).to have_http_status(:created)
+    context 'when submitting valid data' do
+      context 'When submitting all fields' do
+        it 'responds with a ok status' do
+          post(base_path, params:)
+          expect(response).to have_http_status(:ok)
+        end
+
+        it 'responds with a PDF' do
+          post(base_path, params:)
+          expect(response.content_type).to eq('application/pdf')
+        end
       end
+    end
+
+    context 'when submitting valid data without optional fields' do
+      context 'When submitting all fields except claimant' do
+        before do
+          params[:pdf_generator2122].delete(:claimant)
+        end
+
+        it 'responds with a ok status' do
+          post(base_path, params:)
+          expect(response).to have_http_status(:ok)
+        end
 
-      it 'responds with a PDF' do
-        post(base_path, params:)
-        expect(response.content_type).to eq('application/pdf')
+        it 'responds with a PDF' do
+          post(base_path, params:)
+          expect(response.content_type).to eq('application/pdf')
+        end
+      end
+
+      context 'When submitting none of the optional fields' do
+        before do
+          params[:pdf_generator2122][:veteran][:name].delete(:middle)
+          params[:pdf_generator2122][:veteran].delete(:va_file_number)
+          params[:pdf_generator2122][:veteran][:address].delete(:address_line2)
+          params[:pdf_generator2122][:veteran][:address].delete(:zip_code_suffix)
+          params[:pdf_generator2122][:veteran].delete(:phone)
+          params[:pdf_generator2122][:veteran].delete(:service_number)
+          params[:pdf_generator2122].delete(:claimant)
+        end
+
+        it 'responds with a ok status' do
+          post(base_path, params:)
+          expect(response).to have_http_status(:ok)
+        end
+
+        it 'responds with a PDF' do
+          post(base_path, params:)
+          expect(response.content_type).to eq('application/pdf')
+        end
       end
     end
 
     context 'when submitting incomplete data' do
-      it 'responds with an unprocessable entity status' do
-        params[:pdf_generator2122][:organization_name] = nil
-        post(base_path, params:)
-        expect(response).to have_http_status(:unprocessable_entity)
+      context 'when submitting without the organization name' do
```

Comment:
You can DRY this up by moving the following into a `before` block in the `context` block.

```
          params[:pdf_generator2122][:organization_name] = nil
          post(base_path, params:)
 ```

Reply:
Good call, I moved all the setup into `before` blocks.